Switched from log to tracing.

vicky5124 opened the pull request:
This may need a check for places that may be spammy and have instrument, but from what i tested, nothing that spams has instrument with command arguments.

vicky5124 committed:
Switched log to tracing.
	modified:   Cargo.toml
	modified:   src/client/bridge/gateway/shard_manager.rs
	modified:   src/client/bridge/gateway/shard_manager_monitor.rs
	modified:   src/client/bridge/gateway/shard_queuer.rs
	modified:   src/client/bridge/gateway/shard_runner.rs
	modified:   src/client/mod.rs
	modified:   src/framework/standard/help_commands.rs
	modified:   src/gateway/shard.rs
	modified:   src/gateway/ws_client_ext.rs
	modified:   src/http/client.rs
	modified:   src/http/ratelimiting.rs
	modified:   src/internal/ws_impl.rs
	modified:   src/model/channel/reaction.rs
	modified:   src/model/guild/mod.rs
	modified:   src/voice/connection.rs
	modified:   src/voice/streamer.rs
	modified:   src/voice/tasks.rs

vicky5124 committed:
Instrument functions that need it.
	modified:   src/cache/mod.rs
	modified:   src/client/bridge/gateway/shard_manager.rs
	modified:   src/client/bridge/gateway/shard_manager_monitor.rs
	modified:   src/client/bridge/gateway/shard_queuer.rs
	modified:   src/client/bridge/gateway/shard_runner.rs
	modified:   src/client/dispatch.rs
	modified:   src/client/event_handler.rs
	modified:   src/client/mod.rs
	modified:   src/error.rs
	modified:   src/framework/standard/mod.rs
	modified:   src/gateway/shard.rs
	modified:   src/gateway/ws_client_ext.rs
	modified:   src/http/client.rs
	modified:   src/http/ratelimiting.rs
	modified:   src/http/request.rs
	modified:   src/internal/ws_impl.rs
	modified:   src/lib.rs
	modified:   src/utils/mod.rs
	modified:   src/voice/connection.rs
	modified:   src/voice/handler.rs
	modified:   src/voice/streamer.rs
	modified:   src/voice/tasks.rs

vicky5124 committed:
Updated the example to use tracing.
	modified:   examples/README.md
	modified:   examples/e08_env_logging/Cargo.toml
	modified:   examples/e08_env_logging/Makefile.toml
	modified:   examples/e08_env_logging/src/main.rs
	modified:   src/client/bridge/gateway/shard_runner.rs
	modified:   src/client/dispatch.rs

BlackHoleFox commented on examples/e08_env_logging/src/main.rs:
```diff
@@ -26,25 +36,64 @@ impl EventHandler for Handler {
 }
+
+// Currently, the instrument macro doesn't work with commands.
+// if you wish to intrument commands, use it on the before function.
```
Minor typo here:
```suggestion
// if you wish to instrument commands, use it on the before function.
```

vicky5124 committed:
Added comment for intrumenting commands.

Update examples/e08_env_logging/src/main.rs

Co-authored-by: BlackHoleFox <blackholefoxdev@gmail.com>

vicky5124 committed:
Fixed voice logging.
	modified:   src/lib.rs
	modified:   src/voice/connection.rs
	modified:   src/voice/handler.rs
	modified:   src/voice/streamer.rs
	modified:   src/voice/tasks.rs

vicky5124 committed:
Fixed build failure in full feature circumstance.
	modified:   src/lib.rs
